InventoryIndexer/Indexer/SourceItem/SourceItemReindexStrategy.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Indexer\SourceItem;
+
+use Magento\Framework\ObjectManagerInterface;
+use Magento\InventoryIndexer\Model\IndexerConfig;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Receiver of currently active reindex strategy for source items
+ *
+ * @api
+ */
+class SourceItemReindexStrategy
+{
+    /**
+     * @var ObjectManagerInterface
+     */
+    private $objectManager;
+
+    /**
+     * @var array
+     */
+    private $strategies;
+
+    /**
+     * @var IndexerConfig
+     */
+    private $indexerConfig;
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @param IndexerConfig $indexerConfig
+     * @param array $strategies
+     */
+    public function __construct(
+        ObjectManagerInterface $objectManager,
+        IndexerConfig $indexerConfig,
+        $strategies = []
+    ) {
+        $this->objectManager = $objectManager;
+        $this->strategies = $strategies;
+        $this->indexerConfig = $indexerConfig;
+    }
+
+    /**
+     * @param array $sourceItemIds
+     * @return void
+     */
+    public function executeList(array $sourceItemIds) : void
+    {
+        $this->getStrategy()->executeList($sourceItemIds);
+    }
+
+    /**
+     * @return void
+     */
+    public function executeFull() : void
+    {
+        $this->getStrategy()->executeFull();
+    }
+
+    /**
+     * @param int $sourceItemId
+     * @return void
+     */
+    public function executeRow(int $sourceItemId) : void
```

Comment:
do our static tests allow this extra space after closing parenthesis `) : void`

Reply:
Yes, it's acceptable